### PR TITLE
Prepared working solution of Postgres that can be used with integration tests

### PR DESCRIPTION
**Pull request by zabrowarnyrafal**
Prepared working solution for Windows and Linux, compatible with net standard 2.0.

### CODE REVIEW

**Comment by marcinbudny** [comment]
I'm unable to comment on the files in the PR since it's too big. Here are some gathered comments:
* You need to include PostgreSQL license copy into each postgres copy
* The postgres binaries now total at over 400mb. Is it even feasible to put this into nuget package?
* you need to account for package installation location in .net core, which is the user's home folder
* current implementation of disposable pattern is wrong. When called from finalizer, you should not access other reference objects. See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern I think given nature of the library (integration testing tool) you should just go for simple implementation of 'Dispose()' method
* code in constructor of `PostgresRunner` silences exceptions. How will user learn if / what went wrong? * Let's move the code

**Comment by marcinbudny** [comment]
I'm unable to comment on the files in the PR since it's too big. Here are some gathered comments:

* You need to include PostgreSQL license copy into each postgres copy
* The postgres binaries now total at over 400mb. Is it even feasible to put this into nuget package?
* you need to account for package installation location in .net core, which is the user's home folder
* current implementation of disposable pattern is wrong. When called from finalizer, you should not access other reference objects. See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern I think given nature of the library (integration testing tool) you should just go for simple implementation of 'Dispose()' method
* code in constructor of PostgresRunner silences exceptions. How will user learn if / what went wrong? 
	* Let's move the code out of the constructor.
* allow customization of database name in returned connection string
* the `State` is unnecessary, right?
* unique hash would be more unique if just left as a GUID. but isn't port number enough instead of hash? just make sure to clear the dir on startup if it was left on the disk during previous crashed execution

**Comment by zabrowarnyrafal** [comment]
Ad1: https://github.com/bt-skyrise/Postgres2Go/issues/2 (this will be done when code be ready for publishing)
Ad2: indeed; osx binaries cause oversize of tools; for now only win64 and lin64 solution will be prepared and osx solution as more effective managing  Pg binaries will be postponed to one of next releases after nuget package will be published.
Ad3: need to test and fix eventual issue before 0.1.0 release; IMO not need it to be part of this PR
Ad4: this have to be fixed; will be part of this PR
Ad5: diagnostic need to be rethinked; further enhencement - maybe 0.3.0 release
Ad6: not necessary part of PR; this must be definitely done before 0.2.0 release
Ad7: usage need to be rethinked; not need to be change till 0.1.0 release; IMO can be done after 0.3.0 release
Ad8: GUID (or UUID) usage can cause path to temp folder to be too long; 6-characters hash should be enough; port number could be also used;  clearing of directory in test setup procedure must be implemented before 0.1.0 release but not necessary part of this PR


To summarize: ad2, ad4 need to be fixed as a part of this PR. Rest can be done as a different PR.

**Comment by marcinbudny** [comment]
5 and 6 have easy fixes and significant value.
8: Currently paths are generated under `C:\Users\[user]\AppData\Local\Temp`, so you won't exceed MAX_PATH

**Comment by marcinbudny** [comment]
One more comment: use an options object instead of single parameters to configure call to `PostgresRunner.Start` - this way it will be easier to change things in the future and keep backwards compatibility.

**Comment by zabrowarnyrafal** [comment]
Ad1- postponed
Ad2 - done
Ad3- postponed
Ad4 - done
Ad5 - done (but it will be much improved in next releases)
Ad5 - done
Ad6 - done
Ad7 - postponed
Ad8 - postponed
Ad9 - params object - postponed

**Review by marcinbudny — APPROVED** [review]
